### ISSUE TITLE
Adjust the provider plugin gRPC interface to fill out the first-class provider model

### ISSUE THREAD

**Issue by pgavlin** [pulumi/pulumi#1718]
The provider plugin gRPC interface needs to support the following methods/capabilities in order to finish up the first-class provider implementation:
- [x] `CheckConfig` + `DiffConfig` (https://github.com/pulumi/pulumi/pull/2512)
- [ ] `Configure` needs a new version that accepts a property bag and returns outputs
- [ ] `GetPluginInfo` may need a way to indicate that the plugin supports the above extensions

**Comment by lukehoban** [comment]
@pgavlin Could you summarize the user-impacting improvements this will offer?  I'm not yet sure how critical this is to do now vs. later.

**Comment by metral** [comment]
@lukehoban @ellismg any update on which milestone this will be worked on? It's currently a req / blocker for https://github.com/pulumi/pulumi-eks/issues/107

**Comment by lukehoban** [comment]
@pgavlin What remaining work is truly critical here?  What concrete scenarios are broken due to the lack of the remaining pieces of this?

**Comment by lukehoban** [comment]
Closing this out - as @pgavlin indicated there is not required work on the remaining two items.  @pgavlin Feel free to open a new issue with details if there is indeed anything remaining you'd like to track.
